code/modules/events/flutes.dm

```diff
@@ -0,0 +1,142 @@
+//
+//add ability to cure sanity instability with psychatrist?
+//add goals?
+//xoxeyos
+//massive help from Art Prime
+
+//---------
+//UTILITIES
+//---------
+/proc/list_intersection(var/list/A, var/list/B)
+	var/intersection[0]
+	for(var/item in A)
+		if(item in B)
+			intersection.Add(item)
+	return intersection
+
+//----------------------------
+//ROUND EVENT CONTROL "FLUTES"
+//----------------------------
+/datum/round_event_control/flutes
+	name = "Flutes"
+	typepath = /datum/round_event/flutes
+	max_occurrences = 1
+	weight = 1
+	min_players = 30
+
+//--------------------
+//ROUND EVENT "FLUTES"
+//--------------------
+//OBJECT DEF
+/datum/round_event/flutes
+	var/chosen_players[0]
+
+//ROUND START PROC
+//This gets executed at the start of the round
+/datum/round_event/flutes/start()
+	var/list/remaining_players = list_intersection(GLOB.player_list, GLOB.alive_mob_list)
+	var/number_of_players_to_select = round((remaining_players.len / 10), 1)
+	for(var/i; i<=(number_of_players_to_select == 0 ? 1 : number_of_players_to_select); i++)
+		var/player = remaining_players[rand(1, remaining_players.len)]
+		sound_intro(player)
+		pick_flute_scene(player)
+		remaining_players.Remove(player)
+		chosen_players.Add(player)
```

Comment:
This whole proc is a bit weird..

This optimized version does a few things:

1) No need to loop through the lists several times to see where it intersects
2) C style loops (i=start;i>=end;i++) are much less optimized than "in to" loops (i in start to end)
3) We can stop looping through lists once we have our target amount of players
4) Variables names should fit within a couple of characters.... number_of_players_to_select is wayyy too long and verbose

Also remove list_intersection() since you dont need it no more
```suggestion
	var/target_amount = round((remaining_players.len / 10), 1)
	var/list/mob/living/carbon/selected = list()

	for(var/mob/living/carbon/C in GLOB.player_list)
		if(C.stat)
			continue //ignore unconcious/dead players

		sound_intro(C)
		pick_flute_scene(C)
		selected.Add(C)

		if(selected.len >= target_amount)
			break //were done here
```